src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -218,15 +224,17 @@ class AppsResource @Inject() (
   @PUT
   @Timed
   def replaceMultiple(@DefaultValue("false")@QueryParam("force") force: Boolean, body: Array[Byte]): Response = {
-    val updates = Json.parse(body).as[Seq[AppUpdate]]
+    val updates = Json.parse(body).as[Seq[V2AppUpdate]]
     BeanValidation.requireValid(ModelValidation.checkUpdates(updates))
     val version = Timestamp.now()
-    def updateApp(update: AppUpdate, app: AppDefinition): AppDefinition = {
-      update.version.flatMap(v => service.getApp(app.id, v)).orElse(Some(update(app))).getOrElse(app)
+    def updateApp(update: V2AppUpdate, app: AppDefinition): AppDefinition = {
+      val previousVersion = update.version.flatMap(service.getApp(app.id, _))
+      previousVersion.orElse(Some(update(V2AppDefinition(app)).toAppDefinition)).getOrElse(app)
     }
     def updateGroup(root: Group): Group = updates.foldLeft(root) { (group, update) =>
+      val transform = (app: AppDefinition) => updateApp(update, app)
```

Comment:
_Why not a def?_ The value is lifted to a function object when passing it to the higher-order function anyway.
_Why?_ I found the first version confusing to read at first.

I'll change it back.